client/src/main/scala/client/WebSocketClient.scala

```diff
@@ -54,15 +54,15 @@ class WebSocketClient(url: String, timeout: Option[FiniteDuration]) extends Obse
       case NonFatal(e) => Observable.error(e)
     }
 
-    val (inbound, closeConnection: (() => Unit)) = try {
+    val (inbound, closeConnection: (() => Any)) = try {
       val webSocket = new WebSocket(url)
       inboundWrapper(webSocket) -> (() => if (webSocket.readyState <= 1) Try(webSocket.close()))
     } catch {
-      case NonFatal(e) => Observable.error(e) -> ()
+      case NonFatal(e) => Observable.error(e) -> (() => ())
     }
 
     timeout.fold(inbound)(t => inbound.timeout(t))
-      .doOnCanceled(closeConnection)
+      .doOnCanceled(closeConnection())
```

Comment:
As I said, it is ok. The idea of my branch was to remove warnings of scala compiler. If new branch does not bring new warnings, then it is fine. Anyways I will check your branch once it is ready for merge to master